grease/src/Grease/Macaw/ResolveCall.md

```diff
@@ -162,101 +169,180 @@ lookupFunctionHandle bak la halloc arch memory symMap pltStubs dynFunMap funOvs
+             -- or shared library, resolve it...
+             Just pltCallAddr <- Map.lookup pltStubName dynFunMap
+          -> do doLog la $ Diag.PltCall pltStubName funcAddrOff pltCallAddr
+                callConcreteAddr bak la halloc arch memory symMap pltStubs dynFunMap funOvs st mem regs bvMemWord pltCallAddr
+          |  otherwise
+          -> case Map.lookup pltStubName funOvs of
+               -- ...otherwise, if there is an override for the PLT stub,
+               -- use it...
+               Just macawFnOv -> do
+                 logFunctionCall pltStubName
+                 useMacawFunctionOverride bak la halloc arch funOvs macawFnOv st
+               -- ...otherwise, skip the PLT call entirely.
+               Nothing -> do
+                 logFunctionCall pltStubName
+                 skip (PltNoOverride @arch funcAddrOff pltStubName)
+
+     -- Finally, check if this is a function that we should explore, and if
+     -- so, use Macaw's code discovery to do so. See Note [Incremental code
+     -- discovery] in Grease.Macaw.SimulatorState.
+     --
+     -- As a simple heuristic for whether a function is worthy of
+     -- exploration, we check if the segment that the address inhabits is
+     -- executable. This check is important to prevent simulating functions
+     -- that, say, inhabit the .data section (which is common for binaries
+     -- that fail the `in-text` requirement), as Macaw will simply crash
+     -- when simulating them.
+     | Discovery.isExecutableSegOff funcAddrOff -> do
+       (hdl, st') <-
+         discoverFuncAddr la halloc arch memory symMap pltStubs funcAddrOff st
+       logFunctionCall (C.handleName hdl)
+       callHandle hdl st'
+
+     | otherwise ->
+       skip (NotExecutable funcAddrOff)
+  where
+    skip = skipExternalCall la halloc arch st
+    -- Log that we have performed a function call.
+    logFunctionCall fnName = do
+      mbReturnAddr <-
+        (arch ^. archFunctionReturnAddr) bak (arch ^. archVals) regs mem
+      doLog la $ Diag.FunctionCall fnName bvMemWord mbReturnAddr
+
+-- | Treat an external function as a no-op during simulation.
+skipExternalCall ::
+  C.IsSymInterface sym =>
+  GreaseLogAction ->
+  C.HandleAllocator ->
+  ArchContext arch ->
+  C.SimState p sym (Symbolic.MacawExt arch) rtp f a ->
+  SkippedCall arch ->
+  IO ( MacawFnHandle arch
+     , C.SimState p sym (Symbolic.MacawExt arch) rtp f a
+     )
+skipExternalCall la halloc arch st reason = do
+  doLog la $ Diag.SkippedCall reason
+  let funcName = W4.functionNameFromText "_grease_external"
+  handle <- C.mkHandle' halloc funcName (Ctx.Empty Ctx.:> regStructRepr arch) (regStructRepr arch)
+  let override = C.mkOverride' funcName (regStructRepr arch) $ do
+        args <- C.getOverrideArgs
+        let regs' = Ctx.last $ C.regMap args
+        (arch ^. archOffsetStackPointerPostCall) (C.regValue regs')
+  pure $ useFnHandleAndState handle (C.UseOverride override) st
+```
+
+</details>
 
+```haskell
```

Comment:
If we want to proceed with this approach, I would make a new section here describing system calls.